package/MDAnalysis/topology/LAMMPSParser.py

```diff
@@ -265,37 +270,53 @@ def read_DATA_timestep(self, n_atoms, TS_class, TS_kwargs):
 
     def _parse_pos(self, datalines, pos):
         """Strip coordinate info into np array"""
-        for line in datalines:
+        # TODO: could maybe store this from topology parsing?
+        ids = np.zeros(len(pos), dtype=np.int32)
+
+        for i, line in enumerate(datalines):
             line = line.split()
             n = len(line)
-            idx = int(line[0]) - 1
+            ids[i] = line[0]
+
             if n in (7, 10):
-                pos[idx] = line[4:7]
+                pos[i] = line[4:7]
             elif n in (6, 9):
-                pos[idx] = line[3:6]
+                pos[i] = line[3:6]
+    
+        # save to class for vels later
+        self.order = np.argsort(ids)
+        pos = pos[self.order]
+
+        return pos
 
     def _parse_vel(self, datalines, vel):
         """Strip velocity info into np array in place"""
-        for line in datalines:
+        for i, line in enumerate(datalines):
             line = line.split()
-            idx = int(line[0]) - 1
-            vx, vy, vz = np.float64(line[1:4])
-            vel[idx] = vx, vy, vz
+            vel[i] = line[1:4]
 
-    def _parse_bond_section(self, datalines, nentries):
+        vel = vel[self.order]
```

Comment:
At least add a comment that this should exist because the parse_pos function creates it.